src/robot/running/librarykeywordrunner.py

```diff
@@ -104,7 +104,14 @@ def _run_with_output_captured_and_signal_monitor(self, runner, context):
     def _run_with_signal_monitoring(self, runner, context):
         try:
             STOP_SIGNAL_MONITOR.start_running_keyword(context.in_teardown)
-            return runner()
+            runner_result = runner()
+            if (context.asynchronous.is_coroutine(runner_result) and not
+                context.asynchronous.is_loop_running()):
```

Comment:
I believe this violates PEP-8. The second line of the expression should be indented more and, less importantly, `and not` moved there. One easy way to avoid these issues would be adding new helper that checks both is the result a coroutine and is a loop running. I'm just not sure is that worth the effort (or what that helper should be named).